mitosheet/src/components/taskpanes/Graph/GraphSidebar.tsx

```diff
@@ -119,104 +121,72 @@ const GraphSidebar = (props: {
     columnIDsMapArray: ColumnIDsMap[],
     dfNames: string[];
     graphSidebarSheet: number;
-    columnDtypesMap: Record<string, string>;
+    columnDtypesMap: Record<string, string> | undefined;
     mitoAPI: MitoAPI;
     setUIState: React.Dispatch<React.SetStateAction<UIState>>;
-    model_id: string;
-    lastGraphParams: Record<number, GraphParams | undefined>;
-    setLastGraphParams: (sheetIndex: number, graphParams: GraphParams) => void;
+    graphDataJSON: GraphDataJSON
+    lastStepIndex: number
 }): JSX.Element => {
 
-    const [selectedSheetIndex, _setSelectedSheetIndex] = useState(props.graphSidebarSheet);
+    // We keep track of the graph data separately from the backend state so that 
+    // the UI updates imidietly, even though the backend takes a while to process.
+    const [graphParams, setGraphParams] = useState(() => getGraphParams(props.graphDataJSON, props.graphSidebarSheet, props.sheetDataArray))
 
-    // A wrapper around changing the selected sheet index that makes sure
-    // the correct sheet is displayed, and also loads the most recent graph
-    // from this sheet if it exists
-    const setSelectedSheetIndex = (newSelectedSheetIndex: number): void => {
-        const newParams = getGraphParams(props.lastGraphParams, newSelectedSheetIndex, props.sheetDataArray);
-        // Note we update the sheet before the graph parameters
-        _setSelectedSheetIndex(newSelectedSheetIndex);
-        _setGraphParams(newParams)
-        props.setUIState(prevUIState => {
-            return {
-                ...prevUIState,
-                selectedSheetIndex: newSelectedSheetIndex
-            }
-        });
-    }
+    // When updateGraph is changed from false to true, we send a new getGraphMessage with the current graphParams
```

Comment:
Explain that we do this to simulate a callback, but gives us power over what sources of graphParams changing causes the change. Also, note that there is a potential race condition here

Reply:
Note this in the PR description as well since its important